### PR TITLE
Release node v3.5.6

### PR DESCRIPTION
**Pull request by bsudekum**
Going to start off with a prerelease since so much has changed since the last release.

/cc @brunoabinader @tmpsantos @jfirebaugh @ivovandongen

### CODE REVIEW

**Comment by bsudekum** [comment]
Tests and binary creation is failing with

```
ERRO[08:49:38] Failed to get workflow id by pattern, error: no matching workflow found with trigger params: push-branch: , pr-source-branch: , pr-target-branch: , tag: v3.5.6-pre.1 
```

**Comment by bsudekum** [comment]
It looks like circle never picked up the pushed tag so this means no linux binaries.

**Comment by bsudekum** [comment]
@mollymerp @jfirebaugh following instructions found here: https://discuss.circleci.com/t/git-tag-deploys-in-2-0/9493/8

However, putting 

```yml
deployment:
  fake_deploy_for_cci2:
    tag: /.*/
    commands:
      - echo "make tags run in 2.0"
```

At the bottom of the config file results in 

```
Build-agent version 0.0.3793-f06cbf5 (2017-07-26T14:29:53+0000)
Configuration errors: 1 error occurred:

* job 'build' is not found
```
[ref](https://circleci.com/gh/mapbox/mapbox-gl-native/14586)

Any ideas?

**Comment by mollymerp** [comment]
@bsudekum I do remember running into this when configuring ci 2.0 for gl-js, but I fixed it by just making the main job named `build` 😕  https://github.com/mapbox/mapbox-gl-js/pull/4939/files#diff-1d37e48f9ceff6d8030570cd36286a61R4

File: circle.yml [hunk 1]
```diff
@@ -322,8 +327,7 @@ jobs:
       JOBS: 4
       BUILDTYPE: RelWithDebInfo
       WITH_EGL: 1
-      PACKAGE_JSON_VERSION: $(node -e "console.log(require('./package.json').version)")
-      PUBLISH: $([[ "${CIRCLE_BRANCH}" == "node-v${PACKAGE_JSON_VERSION}" ]] && echo true)
```
Comment (bsudekum) [inline circle.yml]: Setting env variables seems to not work which is why I added this logic [`platform/node/scripts/after_success.sh`](https://github.com/mapbox/mapbox-gl-native/pull/9610/files#diff-7f6ebb693582f9045c60fa9dcf7a46efR6)

**Comment by bsudekum** [comment]
@brunoabinader this is ready for review.

File: circle.yml [hunk 2]
```diff
@@ -10,9 +10,18 @@ workflows:
               ignore: master
       - android-debug-arm-v7
       - android-release-all
-      - node4-clang39-release
-      - node6-clang39-release
-      - node6-clang39-debug
+      - node4-clang39-release:
+          filters:
+            tags:
+              only: /node-.*/
```
Comment (bsudekum) [inline circle.yml]: Necessary for tagging releases on circle v2.

Comment (bsudekum) [reply circle.yml]: This filters here, filter tags IN instead of out.

**Pull request merged**